### PR TITLE
[Release PR] Check the VPN egress server dynamically during leak checks

### PR DESCRIPTION
**Pull request by samsymons**
Task/Issue URL: https://app.asana.com/1/137249556945/project/72649045549333/task/1214410332899564?focus=true
Tech Design URL:
CC:

### Description

This PR fixes the egress server check so that stale egress values cannot trigger a false-positive leak check.

### Testing Steps

1. Run the VPN
2. Trigger a rekey event
3. Check that the leak detection service runs and shows a success

### Notes to Reviewer

---
###### Internal references:
[Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f) | [Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552) | [Tech Design Template](https://app.asana.com/0/59792373528535/184709971311943)

---

> [!NOTE]
> **Medium Risk**
> Changes leak-check scheduling and reference-IP selection in the VPN tunnel provider, which could affect when leak checks run and how results are classified. Logic is covered by updated tests but should be validated around rekey/reconnect edge cases.
> 
> **Overview**
> Leak checks now **resolve egress IP/server name dynamically at the start of each check** via an injected async provider, rather than caching and updating a stored `LeakCheckEgressInfo`, and will skip checks when either the tunnel interface or egress reference is unavailable.
> 
> Adds a new leak-check trigger, `rekey`, and updates cooldown logic so `.reassert` and `.rekey` bypass cooldown. `PacketTunnelProvider` now schedules an immediate leak check after `rekey()` *only when the egress server/IP actually changes*, and wide-event schemas/tests are updated to include the new trigger and the provider-based egress behavior.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 49de51b78d9fd264578e2b4f1cd839da9c6082d4. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: SharedPackages/VPN/Sources/VPN/LeakCheck/LeakCheckModels.swift [hunk 1]
```diff
@@ -22,6 +22,7 @@ public enum LeakCheckTrigger: String, Codable {
     case tunnelStart = "tunnel_start"
     case reassert
     case periodic
+    case rekey
```
Comment (samsymons) [inline SharedPackages/VPN/Sources/VPN/LeakCheck/LeakCheckModels.swift:25]: Added a rekey trigger for the leak check. This only runs when rekeying happens and it caused us to change server, otherwise the leak check will run based on the timer as usual.

File: SharedPackages/VPN/Sources/VPN/LeakCheck/VPNLeakCheckService.swift [hunk 2]
```diff
@@ -25,9 +25,10 @@ import PixelKit
 public actor VPNLeakCheckService {
 
     public typealias TunnelInterfaceProvider = @Sendable () async -> NWInterface?
+    public typealias EgressInfoProvider = @Sendable () async -> LeakCheckEgressInfo?
 
     private let configuration: LeakCheckConfiguration
-    private var egressInfo: LeakCheckEgressInfo
+    private let egressInfo: EgressInfoProvider
```
Comment (samsymons) [inline SharedPackages/VPN/Sources/VPN/LeakCheck/VPNLeakCheckService.swift:31]: This is the main fix - the leak check service was relying on being told when the server changed before, and one path was missed. Now it will look up the server info dynamically 100% of the time.

File: SharedPackages/VPN/Sources/VPN/LeakCheck/VPNLeakCheckService.swift [hunk 3]
```diff
@@ -162,7 +164,10 @@ public actor VPNLeakCheckService {
 
         Logger.networkProtectionIPLeakCheck.log("🟢 Starting leak check (trigger: \(trigger.rawValue, privacy: .public))")
 
-        let egressInfoSnapshot = egressInfo
+        guard let egressInfoSnapshot = await egressInfo() else {
+            Logger.networkProtectionIPLeakCheck.log("🔴 Skipping leak check — egress info unavailable (trigger: \(trigger.rawValue, privacy: .public))")
```
Comment (samsymons) [inline SharedPackages/VPN/Sources/VPN/LeakCheck/VPNLeakCheckService.swift:168]: We would benefit from pixels in the else branch of these guard statements - I'll add these in a follow up since they'll need privacy triage.

File: SharedPackages/VPN/Tests/VPNTests/LeakCheck/VPNLeakCheckServiceTests.swift [hunk 4]
```diff
@@ -414,26 +420,27 @@ final class VPNLeakCheckServiceTests: XCTestCase {
 
         async let checkTask: Void = service.runCheck(trigger: .tunnelStart)
         try? await Task.sleep(nanoseconds: 100_000_000)
-        await service.updateEgressInfo(makeEgressInfo(ip: "5.6.7.8"))
+        box.value = makeEgressInfo(ip: "5.6.7.8")
```
Comment (diegoreymendez) [inline SharedPackages/VPN/Tests/VPNTests/LeakCheck/VPNLeakCheckServiceTests.swift:451]: Just my 2 cents... would it make sense to move this to the tunnel interface callback?  I believe this might make it so  we can remove the sleep entirely.

```swift
let newEgress = makeEgressInfo(ip: "5.6.7.8")                                                                                                              
let service = VPNLeakCheckService(                                                                                                                         
      configuration: .default,                                                                                                                               
      egressInfo: { box.value },                                                                                                                             
      tunnelInterface: {                     
          box.value = newEgress                                                                                                                              
          return Self.systemInterface                                                                                           
      },                                     
      httpClient: http,
      stunClient: stun,                                                                                                                                      
      wideEvent: wideEvent
  )              
```

File: SharedPackages/VPN/Sources/VPN/LeakCheck/VPNLeakCheckService.swift [hunk 5]
```diff
@@ -136,8 +134,12 @@ public actor VPNLeakCheckService {
             Logger.networkProtectionIPLeakCheck.log("Skipping leak check — already in flight (trigger: \(trigger.rawValue, privacy: .public))")
             return
         }
+        // `.reassert` and `.rekey` both signal a tunnel path change (failure recovery /
+        // reconfiguration in the first case, server reselection in the second), so they bypass
+        // cooldown and always run — we want to validate the new path immediately.
         if !bypassCooldown,
            trigger != .reassert,
+           trigger != .rekey,
```
Comment (diegoreymendez) [inline SharedPackages/VPN/Sources/VPN/LeakCheck/VPNLeakCheckService.swift:142]: I believe there's no rekey scenario test.  This is really minor, but just flagging it.

**Review by diegoreymendez — APPROVED** [review]
This is great.

```
🟢 Starting leak check service
🟢 Starting leak check (trigger: tunnel_start)
Resolved tunnel interface for leak check: utun280
IP comparison — expected: 149.88.108.18, got ipv4[http: 149.88.108.18, https: 149.88.108.18, stun: 149.88.108.18], ipv6[http: error(POSIXErrorCode(rawValue: 50): Network is down), https: error(POSIXErrorCode(rawValue: 50): Network is down), stun: error(POSIXErrorCode(rawValue: 50): Network is down)]
🟢 Leak check complete (trigger: tunnel_start, status: SUCCESS, latency: 1000ms, ipv4: [http: success, https: success, stun: success], ipv6: [http: success, https: success, stun: success])
🟢 Starting leak check (trigger: rekey)
Resolved tunnel interface for leak check: utun280
IP comparison — expected: 149.88.108.17, got ipv4[http: 149.88.108.17, https: 149.88.108.17, stun: 149.88.108.17], ipv6[http: error(POSIXErrorCode(rawValue: 50): Network is down), https: error(POSIXErrorCode(rawValue: 50): Network is down), stun: error(POSIXErrorCode(rawValue: 50): Network is down)]
🟢 Leak check complete (trigger: rekey, status: SUCCESS, latency: 1500ms, ipv4: [http: success, https: success, stun: success], ipv6: [http: success, https: success, stun: success])
```

File: SharedPackages/VPN/Sources/VPN/LeakCheck/VPNLeakCheckService.swift (hunk 5 above)
Comment (samsymons) [reply SharedPackages/VPN/Sources/VPN/LeakCheck/VPNLeakCheckService.swift:142]: Added a new test - thanks!
